src/ResourceManagement/CosmosDB/Domain/SqlContainer/Definition/IDefinition.cs

```diff
@@ -0,0 +1,186 @@
+        IWithIndexingPolicy<ParentT>,
+        IWithPartitionKey<ParentT>,
+        IWithDefaultTtl<ParentT>,
+        IWithUniqueKeyPolicy<ParentT>,
+        IWithConflictResolutionPolicy<ParentT>,
+        IWithChildResource<ParentT>
+    {
+    }
+
+    /// <summary>
+    /// The stage of a sql container definition allowing to set options.
+    /// </summary>
+    /// <typeparam name="ParentT">The stage of the parent definition to return to after attaching this definition.</typeparam>
+    public interface IWithOptions<ParentT> :
+        HasOptions.Definition.IWithOptions<IWithAttach<ParentT>>
+    {
+    }
+
+    /// <summary>
+    /// The stage of a sql container definition allowing to set through put.
+    /// </summary>
+    /// <typeparam name="ParentT">The stage of the parent definition to return to after attaching this definition.</typeparam>
+    public interface IWithThroughput<ParentT> :
+        HasThroughputSettings.Definition.IWithThroughput<IWithAttach<ParentT>>
+    {
+    }
+
+    /// <summary>
+    /// The stage of a sql container definition allowing to set indexing policy.
+    /// </summary>
+    /// <typeparam name="ParentT">The stage of the parent definition to return to after attaching this definition.</typeparam>
+    public interface IWithIndexingPolicy<ParentT>
+    {
+        /// <summary>
+        /// Specifies the indexing policy.
+        /// </summary>
+        /// <param name="indexingPolicy">The whole object of the indexing policy.</param>
+        /// <returns>The next stage of the definition.</returns>
+        IWithAttach<ParentT> WithIndexingPolicy(Models.IndexingPolicy indexingPolicy);
+
+        /// <summary>
+        /// Starts the definition of the indexing policy.
+        /// </summary>
+        /// <returns>The next stage of the definition.</returns>
+        IndexingPolicy.Definition.IBlank<IWithAttach<ParentT>> DefineIndexingPolicy();
+    }
+
+    /// <summary>
+    /// The stage of a sql container definition allowing to set partition key.
+    /// </summary>
+    /// <typeparam name="ParentT">The stage of the parent definition to return to after attaching this definition.</typeparam>
+    public interface IWithPartitionKey<ParentT>
+    {
+        /// <summary>
+        /// Specifies the container partition key.
+        /// </summary>
+        /// <param name="containerPartitionKey">The whole object of the container partition key.</param>
+        /// <returns>The next stage of the definition.</returns>
+        IWithAttach<ParentT> WithContainerPartitionKey(ContainerPartitionKey containerPartitionKey);
+    }
+
+    /// <summary>
+    /// The stage of a sql container definition allowing to set default ttl.
+    /// </summary>
+    /// <typeparam name="ParentT">The stage of the parent definition to return to after attaching this definition.</typeparam>
+    public interface IWithDefaultTtl<ParentT>
+    {
+        /// <summary>
+        /// Specifies the default ttl.
```

Comment:
TTL? Better to use full name. And in what unit?

Reply:
done, I don't know what is was used to. swagger description was just `Default time to live`